cmd/globals.go

```diff
@@ -123,8 +123,8 @@ var (
 	globalHTTPServerErrorCh = make(chan error)
 	globalOSSignalCh        = make(chan os.Signal, 1)
 
-	// Enable HTTP request/response headers and body logging.
-	globalHTTPTrace bool
+	// File to log HTTP request/response headers and body.
+	globalHTTPTrace *os.File
```

Comment:
I am little worried about keeping *os.File like this and share it with multiple functions, we might get erroneous output. perhaps we can remember the fd() instead and then use os.NewFile() when we need to pass the *os.File

Also during shutdown/restart we should Close() this file i.e we should call globalHTTPTrace.Close()..

Reply:
> I am little worried about keeping *os.File like this and share it with multiple functions, we might get erroneous output.

@harshavardhana can you explain more on the erroneous output and how it can be fixed with fd()?

Reply:
The problem is finalizers which are set on *os.File I am not sure how Go deals with a global variable like this.

So I thought keeping an FD is better so we can create new instances of *os.File as needed.

Reply:
since it will not be garbage collected it will work as expected. If you look at the new code, it has added complexity without adding value

Reply:
That might be true - I guess we need to just close the file properly during stop and restart. We can avoid keeping FD style instead just a global *os.File.

Can you revert to previous but including closing the file upon stop and restart ? @kannapanr